C++ 4.8.5 build failed

When I tried to build on centos 7 with gcc-c++ 4.8.5, it failed with below message:

```
bash-4.2# g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/4.8.5/lto-wrapper
Target: x86_64-redhat-linux
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-bootstrap --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-linker-hash-style=gnu --enable-languages=c,c++,objc,obj-c++,java,fortran,ada,go,lto --enable-plugin --enable-initfini-array --disable-libgcj --with-isl=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/isl-install --with-cloog=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/cloog-install --enable-gnu-indirect-function --with-tune=generic --with-arch_32=x86-64 --build=x86_64-redhat-linux
Thread model: posix
gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC) 
```


```
[ 34%] Building CXX object src/cli/CMakeFiles/appc.dir/ArgumentParser.cpp.o
In file included from /root/code/app-mesh/src/cli/ArgumentParser.cpp:22:0:
/root/code/app-mesh/src/cli/../common/jwt-cpp/jwt.h: In instantiation of 'class jwt::basic_claim<jwt::picojson_traits>':
/usr/include/c++/4.8.2/bits/stl_pair.h:102:11:   required from 'struct std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> >'
/usr/include/c++/4.8.2/type_traits:615:28:   required from 'struct std::__is_destructible_impl<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > >'
/usr/include/c++/4.8.2/type_traits:637:12:   required from 'struct std::__is_destructible_safe<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> >, false, false>'
/usr/include/c++/4.8.2/type_traits:652:12:   required from 'struct std::is_destructible<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > >'
/usr/include/c++/4.8.2/type_traits:116:12:   required from 'struct std::__and_<std::is_destructible<std::pair<const std::basic_string<char, std::char_traits<char>, std::allocator<char> >, jwt::basic_claim<jwt::picojson_traits> > >, std::__is_direct_constructible_impl<std::pair<const std::basic_string<char, std::char_traits<char>, std::allocator<char> >, jwt::basic_claim<jwt::picojson_traits> >, const std::pair<const std::basic_string<char, std::char_traits<char>, std::allocator<char> >, jwt::basic_claim<jwt::picojson_traits> >&> >'
/usr/include/c++/4.8.2/type_traits:817:12:   [ skipping 6 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/c++/4.8.2/type_traits:974:12:   required from 'struct std::is_copy_constructible<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > >'
/usr/include/c++/4.8.2/bits/alloc_traits.h:540:12:   required from 'struct std::__is_copy_insertable<std::allocator<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > > >'
/usr/include/c++/4.8.2/bits/alloc_traits.h:560:63:   required by substitution of 'template<class _Alloc> using __check_copy_constructible = std::__allow_copy_cons<std::__is_copy_insertable<_Alloc>::value> [with _Alloc = std::allocator<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > >]'
/usr/include/c++/4.8.2/bits/unordered_map.h:97:11:   required from 'class std::unordered_map<std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits>, std::hash<std::basic_string<char> >, std::equal_to<std::basic_string<char> >, std::allocator<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > > >'
/root/code/app-mesh/src/cli/../common/jwt-cpp/jwt.h:2453:72:   required from 'class jwt::verifier<jwt::default_clock, jwt::picojson_traits>'
/root/code/app-mesh/src/cli/../common/jwt-cpp/jwt.h:2809:50:   required from here
/root/code/app-mesh/src/cli/../common/jwt-cpp/jwt.h:1779:4: error: function 'jwt::basic_claim<json_traits>::basic_claim(jwt::basic_claim<json_traits>&&) [with json_traits = jwt::picojson_traits]' defaulted on its first declaration with an exception-specification that differs from the implicit declaration 'jwt::basic_claim<jwt::picojson_traits>::basic_claim(jwt::basic_claim<jwt::picojson_traits>&&)'
    basic_claim(basic_claim&&) noexcept = default;
    ^
/root/code/app-mesh/src/cli/../common/jwt-cpp/jwt.h:1781:17: error: function 'jwt::basic_claim<json_traits>& jwt::basic_claim<json_traits>::operator=(jwt::basic_claim<json_traits>&&) [with json_traits = jwt::picojson_traits]' defaulted on its first declaration with an exception-specification that differs from the implicit declaration 'jwt::basic_claim<jwt::picojson_traits>& jwt::basic_claim<jwt::picojson_traits>::operator=(jwt::basic_claim<jwt::picojson_traits>&&)'
    basic_claim& operator=(basic_claim&&) noexcept = default;
                 ^
make[2]: *** [src/cli/CMakeFiles/appc.dir/ArgumentParser.cpp.o] Error 1

```

Write a hello word sample:
```
#include <jwt-cpp/jwt.h>
#include <iostream>
int main(int argc, const char** argv) {
	std::string token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXUyJ9.eyJpc3MiOiJhdXRoMCJ9.AbIJTDMFc7yUa5MhvcP03nJPyCPzZtQcGEp-zWfOkEE";
	auto decoded = jwt::decode(token);
	for(auto& e : decoded.get_payload_claims())
		std::cout << e.first << " = " << e.second.to_json() << std::endl;
}
```

Tried on a clean centos 7 docker container, the issue still exist:
```
$docker run -ti -v `pwd`:`pwd` --rm centos:7 bash
$yum install gcc-c++ openssl-devel -y
```


```
[root@7e64b43c368a test]# g++ --std=c++11 -I. ./jwt.cpp 
In file included from ./jwt.cpp:1:0:
./jwt-cpp/jwt.h: In instantiation of 'class jwt::basic_claim<jwt::picojson_traits>':
/usr/include/c++/4.8.2/bits/stl_pair.h:102:11:   required from 'struct std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> >'
/usr/include/c++/4.8.2/type_traits:615:28:   required from 'struct std::__is_destructible_impl<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > >'
/usr/include/c++/4.8.2/type_traits:637:12:   required from 'struct std::__is_destructible_safe<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> >, false, false>'
/usr/include/c++/4.8.2/type_traits:652:12:   required from 'struct std::is_destructible<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > >'
/usr/include/c++/4.8.2/type_traits:116:12:   required from 'struct std::__and_<std::is_destructible<std::pair<const std::basic_string<char, std::char_traits<char>, std::allocator<char> >, jwt::basic_claim<jwt::picojson_traits> > >, std::__is_direct_constructible_impl<std::pair<const std::basic_string<char, std::char_traits<char>, std::allocator<char> >, jwt::basic_claim<jwt::picojson_traits> >, const std::pair<const std::basic_string<char, std::char_traits<char>, std::allocator<char> >, jwt::basic_claim<jwt::picojson_traits> >&> >'
/usr/include/c++/4.8.2/type_traits:817:12:   [ skipping 6 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/c++/4.8.2/type_traits:974:12:   required from 'struct std::is_copy_constructible<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > >'
/usr/include/c++/4.8.2/bits/alloc_traits.h:540:12:   required from 'struct std::__is_copy_insertable<std::allocator<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > > >'
/usr/include/c++/4.8.2/bits/alloc_traits.h:560:63:   required by substitution of 'template<class _Alloc> using __check_copy_constructible = std::__allow_copy_cons<std::__is_copy_insertable<_Alloc>::value> [with _Alloc = std::allocator<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > >]'
/usr/include/c++/4.8.2/bits/unordered_map.h:97:11:   required from 'class std::unordered_map<std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits>, std::hash<std::basic_string<char> >, std::equal_to<std::basic_string<char> >, std::allocator<std::pair<const std::basic_string<char>, jwt::basic_claim<jwt::picojson_traits> > > >'
./jwt-cpp/jwt.h:2453:72:   required from 'class jwt::verifier<jwt::default_clock, jwt::picojson_traits>'
./jwt-cpp/jwt.h:2809:50:   required from here
./jwt-cpp/jwt.h:1779:4: error: function 'jwt::basic_claim<json_traits>::basic_claim(jwt::basic_claim<json_traits>&&) [with json_traits = jwt::picojson_traits]' defaulted on its first declaration with an exception-specification that differs from the implicit declaration 'jwt::basic_claim<jwt::picojson_traits>::basic_claim(jwt::basic_claim<jwt::picojson_traits>&&)'
    basic_claim(basic_claim&&) noexcept = default;
    ^
./jwt-cpp/jwt.h:1781:17: error: function 'jwt::basic_claim<json_traits>& jwt::basic_claim<json_traits>::operator=(jwt::basic_claim<json_traits>&&) [with json_traits = jwt::picojson_traits]' defaulted on its first declaration with an exception-specification that differs from the implicit declaration 'jwt::basic_claim<jwt::picojson_traits>& jwt::basic_claim<jwt::picojson_traits>::operator=(jwt::basic_claim<jwt::picojson_traits>&&)'
    basic_claim& operator=(basic_claim&&) noexcept = default;

```

--- a/include/jwt-cpp/jwt.h
+++ b/include/jwt-cpp/jwt.h
@@ -1842,9 +1842,9 @@ namespace jwt {
 
 			basic_claim() = default;
 			basic_claim(const basic_claim&) = default;
-			basic_claim(basic_claim&&) noexcept = default;
+			basic_claim(basic_claim&&) = default;
 			basic_claim& operator=(const basic_claim&) = default;
-			basic_claim& operator=(basic_claim&&) noexcept = default;
+			basic_claim& operator=(basic_claim&&) = default;
 			~basic_claim() = default;
 
 			JWT_CLAIM_EXPLICIT basic_claim(typename json_traits::string_type s)

